tests/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/main.yml

```diff
@@ -40,10 +40,16 @@
             selfsigned_digest: sha256
           register: selfsigned_certificate
 
+    # NOTE: 'package' does not work properly with the 'default' docker image, since
+    #       you're in most cases using the wrong Python versions. The 'default' image
+    #       is based on Ubuntu/Debian, so we use `command:` with `apt-get install`.
+    #
+    #- name: 'Install unzip'
+    #  package:
+    #    name: unzip
+    #  when: ansible_distribution != "MacOSX"  # unzip already installed
     - name: 'Install unzip'
-      package:
-        name: unzip
-      when: ansible_distribution != "MacOSX"  # unzip already installed
+      command: apt-get install unzip
```

Comment:
I had the done the same (but in `runme.sh`) in #2 , but mostly as a workaround to get to the next error.

My main concern with this is that it forces the tests to only ever run on debian/ubuntu. I get that's all we're doing in CI now, not sure that will always be the case; and it kind of precludes anyone running the tests on their own outside of debian derivatives too, doesn't it?

Once the rest were working, I was thinking of going back and maybe adding a task conditional on debian/ubuntu that tries to implement some of the workarounds in this post: https://stackoverflow.com/a/53153996/3905079

Reply:
I think it will be a lot easier to fix properly once something like ansible/ansible#72430 is available. The `default` docker container is sufficiently different from the OS containers that could also be used. Once it is possible to distinguish them, one can use `command: apt-get install ...` for the default container and the `package` module in other cases.

@mattclay this is one case where knowing the target seems to be not enough, since there is no target. All tests are always run, the only difference is which values for `--docker` and `--python` are passed to `ansible-test`.

Reply:
Echoing the concern about this failing on distros other than debian/ubuntu but we can live with this for now.

Reply:
Why does the `package` module not work in this case?

Reply:
@mattclay `package` tries to load the `apt` python module, which fails. It has a fallback for that, whereby it installs the required `python3-apt` system package then tries again. However what happens under the covers is that apt says "`python3-apt` is already installed", which it is. But if I understand it correctly, the reason it can't be loaded is that you need these apt bindings for every version of python, and the one installed is for the system version of python (which is 3.6), so in 3.6 runs it all works fine, but if the run uses any other version of python, it will fail, and there's no simple solution. It seems like what people usually do for this situation is symlink the installed library to the other python version folders. There are a number of such answers described in this SO question, here's an answer I thought explained it succinctly: https://stackoverflow.com/a/53153996/3905079

Reply:
This will obviously be a larger problem with the Python 3.8+ controller change that's slated for 2.11- that won't affect remotes, but modules local to the controller will inherit the controller Python for executing them, and run afoul of this issue. It's most noticeable on RHELish things that need selinux and/or yum/dnf Python libs, but there are other places (as you've found).

We've got several possible ways of handling this- the most likely candidate is "module respawn"- basically, for modules that know they need a weird library that's only available under a "system Python" like this, there'd be a standard module_util that allows the module to do whatever logic it wants to determine the right Python interpreter, then transparently respawn the module under that interpreter as a child process and pass the results through the original module process (which would just act as a shim to smuggle the results when running in respawn mode).

I had a working prototype of it a couple years ago (where it was trivial), but that was before AnsiballZ had switched to using runpy, which complicates a couple of the details of the respawn. It's a problem we have to solve for 2,11, anyway, though I doubt we'll be backporting it. Short-term, just manually setting the interpreter in a task var to the one that has the necessary libs for on-controller tests is probably the path of least resistance.

Reply:
thanks @nitzmahone , I think I'll take a stab at doing that (in a new PR): putting the `package` task back but passing in a desired "system" python to use on that task.

Reply:
For anyone who wants to follows along: https://github.com/ansible-collections/community.hashi_vault/pull/6

The above change seemed to work fine for Python 3, in 2.7 the tests all run but fail at the end complaining that the tests modified the environment. Will look more into that.